src/fprime/fbuild/interaction.py

```diff
@@ -216,3 +279,194 @@ def new_component(
     except OSError as ose:
         print("[ERROR] {}".format(ose))
     return 1
+
+
+def is_valid_name(word):
+    invalid_characters = [
+        "#",
+        "%",
+        "&",
+        "{",
+        "}",
+        "/",
+        "\\",
+        "<",
+        ">",
+        "*",
+        "?",
+        " ",
+        "$",
+        "!",
+        "'",
+        '"',
+        ":",
+        "@",
+        "+",
+        "`",
+        "|",
+        "=",
+    ]
+    for char in invalid_characters:
+        if char in word:
+            return char
+    return "valid"
+
+
+def get_valid_input(prompt):
+    valid_name = False
+    while not valid_name:
+        name = input(prompt)
+        char = is_valid_name(name)
+        if char != "valid":
+            print("'" + char + "' is not a valid character. Enter a new port name:")
+        else:
+            valid_name = True
+    return name
+
+
+def get_port_input(namespace):
+    # Gather inputs to use as context for the port template
+    defaults = {
+        "port_name": "ExamplePort",
+        "short_description": "Example usage of port",
+        "dir_name": ".",
+        "namespace": namespace,
+        "arg_list": [],
+    }
+    args_done = False
+    arg_list = []
+    port_name = get_valid_input("Port Name [{}]: ".format(defaults["port_name"]))
+    short_description = input(
+        "Short Description [{}]: ".format(defaults["short_description"])
+    )
+    dir_name = get_valid_input("Directory Name [{}]: ".format(defaults["dir_name"]))
+    namespace = get_valid_input("Port Namespace [{}]: ".format(defaults["namespace"]))
+    while not args_done:
+        if arg_list == []:
+            add_arg = confirm("Would you like to add an argument?: ")
+        else:
+            add_arg = confirm("Would you like to add another argument?: ")
+        if add_arg:
+            arg_name = input("Argument name: ")
+            arg_type = input("Argument type: ")
```

Comment:
It might be nice to prompt a user to pick from a list of default types: U8/I8/F32/F64/BOOL/etc. If a user wants to use a custom type there could be an other type that prompts for a raw string.

I think this could be useful for users unfamiliar with F' to know what the valid options for argument type are. What do you think?

Reply:
Yeah, I like that idea. How many types do we want to list before you think it would be too many options?

Reply:
I'd list them all, since only listing some of them would cause confusion as well. I think it's about a dozen options, so hopefully we can fit it in a couple lines.

Reply:
What about custom types?

Reply:
What if we just include the default types in the prompt, but then allow the user to enter whatever type they want? That way new users would still be able to see what some valid options are.

Reply:
Another option would be to have a list of types and then an `other` option. If the user selects other, prompt the user to enter their type. This protects against typos but may be kinda overkill. Your recommendation is a good approach as well, go with whichever you prefer.